opm/output/eclipse/EclipseWriter.cpp

```diff
@@ -296,277 +196,151 @@ class Restart : private boost::noncopyable
     }
 
-    // no inline implementation of these two methods since they depend
-    // on the classes defined in the following.
-
-    // add rate variables for each of the well in the input file
-    void addAllWells(Opm::EclipseStateConstPtr eclipseState,
-                     const PhaseUsage& uses);
-    void writeTimeStep(int writeStepIdx,
-                       const SimulatorTimerInterface& timer,
-                       const WellState& wellState);
+    ecl_rst_file_type* ertHandle() { return this->restart.ertHandle(); }
 
-    ecl_sum_type *ertHandle() const
-    { return ertHandle_; }
+    ~Solution() { ecl_rst_file_end_solution( this->restart.ertHandle() ); }
 
 private:
-    ecl_sum_type *ertHandle_;
-
-    Opm::EclipseStateConstPtr eclipseState_;
-    SummaryReportVarCollection summaryReportVars_;
+    Restart& restart;
 };
 
-class SummaryTimeStep : private boost::noncopyable
-{
-public:
-    SummaryTimeStep(Summary& summaryHandle,
-                    int writeStepIdx,
-                    const SimulatorTimerInterface &timer)
-    {
-        ertHandle_ = ecl_sum_add_tstep(summaryHandle.ertHandle(),
-                                       writeStepIdx,
-                                       timer.simulationTimeElapsed());
-    }
-
-    // no destructor in this class as ERT takes care of freeing the
-    // handle as part of freeing the solution handle!
-
-    ecl_sum_tstep_type *ertHandle() const
-    { return ertHandle_; };
-
-private:
-    ecl_sum_tstep_type *ertHandle_;
-};
-
-
 /**
  * Initialization file which contains static properties (such as
  * porosity and permeability) for the simulation field.
  */
-class Init : private boost::noncopyable
-{
+class Init {
 public:
-    Init(const std::string& outputDir,
-         const std::string& baseName,
-         int writeStepIdx,
-         IOConfigConstPtr ioConfig) : egridFileName_(outputDir,
-                                                     baseName,
-                                                     ECL_EGRID_FILE,
-                                                     writeStepIdx,
-                                                     ioConfig->getFMTOUT())
-    {
-        bool formatted = ioConfig->getFMTOUT();
-
-        FileName initFileName(outputDir,
-                              baseName,
-                              ECL_INIT_FILE,
-                              writeStepIdx,
-                              formatted);
-
-        ertHandle_ = fortio_open_writer(initFileName.ertHandle(),
-                                        formatted,
-                                        ECL_ENDIAN_FLIP);
-    }
+    Init( const std::string& outputDir,
+          const std::string& baseName,
+          int writeStepIdx,
+          const IOConfig& ioConfig ) :
+        egrid( outputDir,
+               baseName,
+               ECL_EGRID_FILE,
+               writeStepIdx,
+               ioConfig.getFMTOUT() ),
+        init( fortio_open_writer(
```

Comment:
Here you can look at: https://github.com/Ensembles/ert/blob/master/devel/libeclxx/include/ert/ecl/FortIO.hpp#L35